src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,277 @@
+        System.out.print(horizontalLine);
+    }
+
+    /**
+     * Appends a task to the task array
+     *
+     * @param task The task inputted by the user
+     */
+    private static void append(Task task) {
+        taskStorage.appendTask(task);
+        System.out.print(horizontalLine + "YOU WANT TO " + task + "?\nSURE, WHATEVER.\n" + horizontalLine);
+    }
+
+    /**
+     * Converts the string into a duke.ToDo object
+     * then appends it to the task array
+     *
+     * @param task description of task
+     */
+    private static void appendToDo(String task) {
+        try {
+            append(new ToDo(Parser.parseToDo(task)));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.print(horizontalLine +
+                    "WRONG FORMAT FOOL!!! IT'S:\n" +
+                    "todo {task}\n" +
+                    horizontalLine);
+        }
+    }
+
+    /**
+     * Converts the string into a duke.Deadline object
+     * then appends it to the task array
+     *
+     * @param task description of task with 'by' time
+     */
+    private static void appendDeadline(String task) {
+        try {
+            String[] parsedDeadline = Parser.parseDeadline(task);
+            append(new Deadline(parsedDeadline[0], parsedDeadline[1]));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.print(horizontalLine +
+                    "WRONG FORMAT FOOL!!! IT'S:\n" +
+                    "deadline {task} /by {time}\n" +
+                    horizontalLine);
+        } catch (DateTimeParseException e) {
+            System.out.print(horizontalLine + "Date format should be yyyy-mm-dd\n"
+                    + horizontalLine);
+        }
+    }
+
+    /**
+     * Converts the string into an duke.Event object
+     * then appends it to the task array
+     *
+     * @param task description of task with 'from' time and 'to' time
+     */
+    private static void appendEvent(String task) {
+        try {
+            String[] parsedEvent = Parser.parseEvent(task);
+            append(new Event(parsedEvent[0], parsedEvent[1], parsedEvent[2]));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.print(horizontalLine +
+                    "WRONG FORMAT FOOL!!! IT'S:\n" +
+                    "event {task} /from {time} /to {time}\n" +
+                    horizontalLine);
+        } catch (DateTimeParseException e) {
+            System.out.print(horizontalLine + "Date format should be yyyy-mm-dd\n"
+                    + horizontalLine);
+        }
+    }
+    /**
+     * Attempts to mark a task in the task array
+     *
+     * @param toMark
+     */
+    private static void mark(String toMark) {
+        System.out.print(horizontalLine);
+        try {
+            Task task = taskStorage.get(Parser.parseMark(toMark));
+            if (task == null) throw new NullPointerException();
+            if (task.isDone) throw new IllegalArgumentException();
```

Comment:
What do you think of doing this instead?
```suggestion
            if (task.isDone) {
              throw new IllegalArgumentException();
            }
```